benchmark/single-source/SetTests.swift

```diff
@@ -55,6 +55,36 @@ let setQ: Set<Int> = {
   return set
 }()
 
+// Construction of empty array.
+let arrayE: Array<Int> = []
+let arrayOE: Array<Box<Int>> = []
+
+// Construction kit for arrays with 25% overlap
+let arrayAB = Array(0 ..< size)                              //   0 ..< 400
+let arrayCD = Array(size ..< 2 * size)                       // 400 ..< 800
+let arrayBC = Array(size - quarter ..< 2 * size - quarter)   // 300 ..< 700
+let arrayB = Array(size - quarter ..< size)                  // 300 ..< 400
+
+let arrayOAB = arrayAB.map(Box.init)
+let arrayOCD = arrayCD.map(Box.init)
+let arrayOBC = arrayBC.map(Box.init)
+let arrayOB = arrayB.map(Box.init)
+
+// Construction kit for arrays with 50% overlap
+let arrayXY = Array(0 ..< size)           //   0 ..< 400
+let arrayYZ = Array(half ..< size + half) // 200 ..< 600
+let arrayY = Array(half ..< size)         // 200 ..< 400
+
+// Two arrays with 100% overlap, but different bucket counts (let's not make it
+// too easy...)
+let arrayP = Array(0 ..< size)
+let arrayQ: Array<Int> = {
+  var array = Array(0 ..< size)
+  array.shuffle()
```

Comment:
When I used `reserveCapacity`, the array wasn't random, so I use `shuffle` instead.
Is it the best way for this?